storage/gcp/bucket/variables.tf

```diff
@@ -0,0 +1,162 @@
+variable "bucket_name" {
+  description = "The name of the bucket."
+  type        = string
+}
+
+variable "alternative_location" {
+  description = "Alternative location for the bucket, by default it will use the project region."
+  type        = string
+  default     = null
+}
+
+variable "force_destroy" {
+  description = "When deleting a bucket, this boolean option will delete all contained objects."
+  type        = bool
+  default     = false
+}
+
+variable "storage_class" {
+  description = "The Storage Class of the new bucket. Supported values include: STANDARD (default), MULTI_REGIONAL, REGIONAL, NEARLINE, COLDLINE, ARCHIVE."
+  type        = string
+  default     = "STANDARD"
+}
+
+variable "autoclass" {
+  description = "The bucket's Autoclass configuration."
+  type        = object({
+    enabled = bool
+  }) 
+  default     = null
+}
+
+variable "lifecycle_rule" {
+  description = "The bucket's Lifecycle Rules configuration."
+  type        = object({
+    action        = object({
+      type          = string
+      storage_class = string
+    })
+    condition = object ({
+      age                        = optional(number)
+      created_before             = optional(string)
+      with_state                 = optional(string)
+      matches_storage_class      = optional(list(string))
+      matches_prefix             = optional(list(string))
+      matches_suffix             = optional(list(string))
+      num_newer_versions         = optional(number)
+      custom_time_before         = optional(string)
+      days_since_custom_time     = optional(string)
+      days_since_noncurrent_time = optional(string)
+      noncurrent_time_before     = optional(string)
```

Comment:
Until now, we required terraform 1.0. The use of optionals would break this compatibility as it requires terraform 1.3.